packit/utils/repo.py

```diff
@@ -542,3 +545,49 @@ def get_commit_message_from_action(
     # it is later reconstructed in a generic way for both defaults and override,
     # so we don't care about the whitespace at the beginning and the end
     return title.strip(), description.strip()
+
+
+def get_tag_link(git_url: str, upstream_tag: str) -> str:
+    """
+    Get link to the tag of a Git repo.
+    """
+    link = ""
+    git_repo = parse_git_repo(git_url)
+    if not git_repo:
+        return ""
+
+    forge = git_repo.hostname
+    if not forge:
+        return ""
+
+    if forge == "github.com":
+        link = f"{git_url}/releases/tag/{upstream_tag}"
+    # GitLab or GitLab instances (e.g. gitlab.gnome.org)
+    elif "gitlab" in forge:
+        link = f"{git_url}/-/tags/{upstream_tag}"
+
+    return link
+
+
+def get_commit_link(git_url: str, upstream_commit: str) -> str:
+    """
+    Get link to the commit of a Git repo.
+    """
+    link = ""
+    git_repo = parse_git_repo(git_url)
+    if not git_repo:
+        return ""
+
+    forge = git_repo.hostname
+    if not forge:
+        return ""
+
+    if forge == "github.com":
+        link = f"{git_url}/commit/{upstream_commit}"
+    # GitLab or GitLab instances (e.g. gitlab.gnome.org)
+    elif "gitlab" in forge:
+        link = f"{git_url}/-/commit/{upstream_commit}"
+    elif forge == "pagure.io":
+        link = f"{git_url}/c/{upstream_commit}"
+
+    return link
```

Comment:
with the current implementation, it didn't make that much sense to move this to OGR to me as it doesn't touch API..

> I could have easily missed something, but can't we work through OGR where possible similarly to the issue_repository?

this relates and would go against packit/packit-service#1907

Reply:
> My point is that we have a very similar code in OGR (e.g. [here](https://github.com/packit/ogr/blob/6a410f63fd1d6c2c9d50ee3abe36b9b21e64d4f0/ogr/services/gitlab/release.py#L34-L36))

I see, but the code there works with a release obtained via API, meanwhile here we are really working only with URLs and do not want the interaction with API at all. But sure, I can move it.

> Regarding the https://github.com/packit/packit-service/issues/1907, it's a question if we want to make it API-independent for all (and remove functionality like issue creation or release description) 

Well, the issue creation is a little bit different, the project doesn't have to be the upstream project. 

> or do API calls where possible and have a new workflow for non-supported ones.

With the mentioned issue what we wanted to achieve was that we can support any upstream without the need of creating tokens on some new Git forges beforehand, so I do not think we should go this way since the URLs are really static.

Reply:
> we are really working only with URLs and do not want the interaction with API at all. But sure, I can move it.

That's the same as with the `parsing.py` and a `RepoUrl` class we already use in Packit. 

> With the mentioned issue what we wanted to achieve was that we can support any upstream without the need of creating tokens on some new Git forges beforehand, so I do not think we should go this way since the URLs are really static.

My point (meant more as a concern/question) was if it makes sense to have a static URL transformation when we know we can get it from some library (e.g. in the case of GitHub, it's for sure). But I agree it's much simpler to have a static mapping for all than to have multiple ways to get that info.  So, +1 for the current approach.